File: backend/app/api/schemas/api_key.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from uuid import UUID

class ApiKeyCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    allowed_domains: List[str] = []

class ApiKeyUpdate(BaseModel):
    name: Optional[str] = None
    allowed_domains: Optional[List[str]] = None
    is_active: Optional[bool] = None

class ApiKeyResponse(BaseModel):
    id: UUID
    workspace_id: UUID
    name: str
    key_prefix: str
    allowed_domains: Optional[List[str]]
    is_active: bool
    requests_count: int
    last_used_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True

class ApiKeyGenerated(ApiKeyResponse):
    full_key: str  # Only returned once upon creation

File: frontend/src/lib/api-key.ts
import api from './api';

export interface ApiKey {
  id: string;
  workspace_id: string;
  name: string;
  key_prefix: string;
  allowed_domains: string[] | null;
  is_active: boolean;
  requests_count: number;
  last_used_at: string | null;
  created_at: string;
}

export interface ApiKeyCreate {
  name: string;
  allowed_domains?: string[];
}

export interface ApiKeyUpdate {
  name?: string;
  allowed_domains?: string[];
  is_active?: boolean;
}

export interface ApiKeyGenerated extends ApiKey {
  full_key: string;
}

export const apiKeyApi = {
  create: (workspaceId: string, data: ApiKeyCreate) => 
    api.post<ApiKeyGenerated>(`/workspaces/${workspaceId}/api-keys`, data),

  getAll: (workspaceId: string) => 
    api.get<ApiKey[]>(`/workspaces/${workspaceId}/api-keys`),

  update: (workspaceId: string, keyId: string, data: ApiKeyUpdate) => 
    api.patch<ApiKey>(`/workspaces/${workspaceId}/api-keys/${keyId}`, data),

  revoke: (workspaceId: string, keyId: string) => 
    api.delete(`/workspaces/${workspaceId}/api-keys/${keyId}`),
};
